sdk/typescript/src/keypairs/passkey/keypair.ts

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { toBase64, toHex } from '@mysten/bcs';
+import { secp256r1 } from '@noble/curves/p256';
+import { blake2b } from '@noble/hashes/blake2b';
+import { randomBytes } from '@noble/hashes/utils';
+import type {
+	AuthenticationCredential,
+	RegistrationCredential,
+} from '@simplewebauthn/typescript-types';
+
+import { PasskeyAuthenticator } from '../../bcs/bcs.js';
+import type { IntentScope, SignatureWithBytes } from '../../cryptography/index.js';
+import { messageWithIntent, SIGNATURE_SCHEME_TO_FLAG, Signer } from '../../cryptography/index.js';
+import type { PublicKey } from '../../cryptography/publickey.js';
+import type { SignatureScheme } from '../../cryptography/signature-scheme.js';
+import {
+	parseDerSPKI,
+	PASSKEY_PUBLIC_KEY_SIZE,
+	PASSKEY_SIGNATURE_SIZE,
+	PasskeyPublicKey,
+} from './publickey.js';
+
+/**
+ * A passkey signer used for signing transactions. This is a client side implementation for [SIP-9](https://github.com/sui-foundation/sips/blob/main/sips/sip-9.md).
+ */
+export class PasskeyKeypair extends Signer {
+	private publicKey: Uint8Array;
+
+	/**
+	 * Get the key scheme of passkey,
+	 */
+	getKeyScheme(): SignatureScheme {
+		return 'Passkey';
+	}
+
+	/**
+	 * Creates an instance of Passkey signer. It's expected to call the static `getPasskeyInstance` method to create an instance.
+	 * For example:
+	 * ```
+	 * const signer = await AwsKmsSigner.getPasskeyInstance();
+	 * ```
+	 */
+	constructor(publicKey: Uint8Array) {
+		super();
+		this.publicKey = publicKey;
+	}
+
+	/**
+	 * Creates an instance of Passkey signer invoking the passkey from navigator.
+	 */
+	static async getPasskeyInstance(): Promise<PasskeyKeypair> {
```

Comment:
Is it expected that we will only ever be able to have 1 passkey per origin?

Reply:
yes, passkey is unique and doesn't change with origin (the initiating website)